src/core/annotation.js

```diff
@@ -563,6 +566,21 @@ class Annotation {
   async save(evaluator, task, annotationStorage) {
     return null;
   }
+
+  /**
+   * Reset the annotation.
+   *
+   * This involves resetting the various streams that are either cached on the
+   * annotation instance or created during its construction.
+   *
+   * @public
+   * @memberof Annotation
+   */
+  reset() {
```

Comment:
One thing that slightly worries me here, with respect to any future code changes, is the need to remember to *manually* add these streams in order for things to work correctly.

To reduce the risk of future bugs, can we at least ensure that `this.appearance` will always be reset here? Something like the following may not be perfect, but should at least help prevent trivial bugs during development:
```suggestion
  reset() {
    if (
      (typeof PDFJSDev === "undefined" ||
        PDFJSDev.test("!PRODUCTION || TESTING")) &&
      this.appearance && !this._streams.includes(this.appearance)
    ) {
      unreachable(/* Error message here... */);
    }
```

Reply:
I think that's a valid concern and we might be able to improve this by unconditionally including `this.appearance` so we don't need to do it at the various call sites.

Reply:
I pushed a new version that centralizes the `this.appearance` pushing in the `Annotation` constructor as much as possible instead of in various places. Would you still like to have this `PDFJSDev` addition or does the current solution suffice?

Reply:
> I pushed a new version that centralizes the `this.appearance` pushing in the `Annotation` constructor as much as possible instead of in various places.

While this obviously *reduces* the risk of future issues, as far as I can tell it doesn't completely eliminate it.

> Would you still like to have this `PDFJSDev` addition or does the current solution suffice?

Yes, please add those checks as well here!
They'll *only* apply in non-PRODUCTION or TESTING builds, i.e. won't affect the run-time in releases, and seem like an easy way to prevent future bugs.

Reply:
Done in the new commit, including the FBF test which failed on master but passes with this patch applied.